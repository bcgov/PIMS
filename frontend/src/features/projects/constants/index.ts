export * from './AgencyResponses';
export * from './DisposalWorkflows';
export * from './DisposeWorkflowStatus';
export * from './ProjectWorkflowComponents';
export * from './ReviewWorkflowStatus';
export * from './SPPApprovalTabs';
