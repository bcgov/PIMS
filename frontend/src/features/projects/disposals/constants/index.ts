export * from './defaultFilter';
export * from './defaultProjectForm';
