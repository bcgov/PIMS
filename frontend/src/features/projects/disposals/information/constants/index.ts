export * from './AddPropertyColumns';
export * from './PropertyColumns';
