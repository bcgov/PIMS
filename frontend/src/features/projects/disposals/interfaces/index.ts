export * from './IProjectAgencyResponseForm';
export * from './IProjectForm';
export * from './IProjectNoteForm';
export * from './IProjectPropertyFilter';
export * from './IProjectPropertyForm';
export * from './IProjectTaskForm';
