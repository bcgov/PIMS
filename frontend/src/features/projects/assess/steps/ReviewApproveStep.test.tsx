import React from 'react';
import * as reducerTypes from 'constants/reducerTypes';
import { createMemoryHistory } from 'history';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { IProject, IProjectTask } from '../../common';
import { ITask, ReviewWorkflowStatus } from '../../common/interfaces';
import { ProjectActions } from 'constants/actionTypes';
import { render } from '@testing-library/react';
import { useKeycloak } from '@react-keycloak/web';
import { Claims } from 'constants/claims';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import { ReviewApproveStep } from '..';
import { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import { Button } from 'react-bootstrap';
import GenericModal from 'components/common/GenericModal';
import { IProperty } from 'features/properties/list/interfaces';
import { act } from 'react-dom/test-utils';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('@react-keycloak/web');
const mockKeycloak = (claims: string[]) => {
  (useKeycloak as jest.Mock).mockReturnValue({
    keycloak: {
      userInfo: {
        agencies: [1],
        roles: claims,
      },
      subject: 'test',
    },
  });
};

const mockStore = configureMockStore([thunk]);
const history = createMemoryHistory();
const mockAxios = new MockAdapter(axios);

const mockTasks: IProjectTask[] = [
  {
    projectNumber: 123,
    taskId: 1,
    isOptional: false,
    isCompleted: true,
    name: 'task-0',
    description: 'one',
    taskType: 1,
    sortOrder: 0,
    completedOn: new Date(),
    statusId: 0,
    statusCode: ReviewWorkflowStatus.PropertyReview,
  },
  {
    projectNumber: 123,
    taskId: 2,
    isOptional: false,
    isCompleted: true,
    name: 'task-1',
    description: 'two',
    taskType: 1,
    sortOrder: 0,
    completedOn: new Date(),
    statusId: 0,
    statusCode: ReviewWorkflowStatus.DocumentReview,
  },
];

const incompleteTask: IProjectTask[] = [
  {
    projectNumber: 123,
    taskId: 1,
    isOptional: false,
    isCompleted: false,
    name: 'task-0',
    description: 'one',
    taskType: 1,
    sortOrder: 0,
    completedOn: new Date(),
    statusId: 0,
    statusCode: ReviewWorkflowStatus.PropertyReview,
  },
];

const mockProperty: IProperty = {
  id: 0,
  propertyTypeId: 0,
  propertyType: 'Land',
  latitude: 23,
  longitude: 23,
  pid: '123-123-123',
  classificationId: 2,
  classification: 'Surplus Active',
  description: 'test',
  isSensitive: false,
  agencyId: 0,
  agency: 'test',
  agencyCode: 'TST',
  address: '1234 Test St',
  addressId: 1,
  administrativeArea: 'Victoria',
  province: 'BC',
  postal: 'A1A 1A1',
  market: 123,
  netBook: 223,
  assessed: 123,
  appraised: 1000,
  landArea: 123,
  landLegalDescription: 'test',
};

const mockProject = (tasks: any) => {
  return {
    projectNumber: 'test-01',
    name: 'my project',
    description: 'my project description',
    privateNote: 'private note',
    publicNote: 'public note',
    properties: [mockProperty],
    agencyId: 1,
    statusId: 0,
    statusCode: ReviewWorkflowStatus.PropertyReview,
    exemptionRequested: true,
    exemptionRationale: 'rationale',
    tierLevelId: 1,
    tasks: tasks,
    note: 'my notes',
    id: 1,
    fiscalYear: 2020,
    projectAgencyResponses: [],
    notes: [],
  } as any;
};

export const tasks: ITask[] = [
  {
    taskId: 1,
    name: 'task-0',
    sortOrder: 0,
    description: 'test',
    taskType: 1,
  },
  {
    taskId: 2,
    name: 'task-1',
    sortOrder: 0,
    description: 'test',
    taskType: 1,
  },
];

const store = (project: IProject) => {
  return mockStore({
    [reducerTypes.LOOKUP_CODE]: { lookupCodes: [] },
    [reducerTypes.ProjectReducers.PROJECT]: { project },
    [reducerTypes.ProjectReducers.TASKS]: tasks,
    [reducerTypes.NETWORK]: {
      [ProjectActions.GET_PROJECT]: {},
    },
  });
};

const getReviewApproveStep = (storeOverride?: any) => (
  <Provider store={storeOverride ?? store(mockProject(mockTasks))}>
    <Router history={history}>
      <ReviewApproveStep />
    </Router>
  </Provider>
);

describe('Review Approve Step', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  beforeEach(() => {
    mockAxios.onAny().reply(200, {});
    mockKeycloak([]);
  });
  it('renders correctly', () => {
    act(() => {
      const { container } = render(getReviewApproveStep());
      expect(container.firstChild).toMatchSnapshot();
    });
  });
  it('edit button is visible when user has correct claims', () => {
    mockKeycloak([Claims.ADMIN_PROJECTS]);
    act(() => {
      const { getByText } = render(getReviewApproveStep());
      const editButton = getByText(/Edit/);
      expect(editButton).toBeTruthy();
    });
  });
  it('edit button is not visible when user does not have required agency/claims', () => {
    act(() => {
      const { queryByText } = render(getReviewApproveStep());
      const editButton = queryByText(/Edit/);
      expect(editButton).toBeFalsy();
    });
  });

  it('Review step submits correctly', async done => {
    mockAxios.reset();
    (useKeycloak as jest.Mock).mockReturnValue({
      keycloak: {
        userInfo: {
          agencies: [1],
          roles: Claims.ADMIN_PROJECTS,
        },
        subject: 'test',
      },
    });
    let component: any;
    await act(async () => {
      component = mount(getReviewApproveStep(store(mockProject(mockTasks))));
      const button = component.findWhere((node: { type: () => any; text: () => string }) => {
        return node.type() === Button && node.text() === 'Save';
      });

      mockAxios
        .onPut()
        .reply((config: any) => {
          done();
          return [200, Promise.resolve({})];
        })
        .onAny()
        .reply((config: any) => {
          return [200, Promise.resolve({})];
        });

      button.simulate('click');
    });
  });
});

describe('Review approve modal behaviour', () => {
  it('confirmation popup does not appear when there are incomplete tasks', async () => {
    (useKeycloak as jest.Mock).mockReturnValue({
      keycloak: {
        userInfo: {
          agencies: [1],
          roles: Claims.ADMIN_PROJECTS,
        },
        subject: 'test',
      },
    });
    await act(async () => {
      const component = mount(getReviewApproveStep(store(mockProject(incompleteTask))));
      const button = component.findWhere((node: { type: () => any; text: () => string }) => {
        return node.type() === Button && node.text() === 'Approve';
      });

      button.simulate('click');

      return Promise.resolve().then(() => {
        expect(component.find(GenericModal)).toHaveLength(0);
      });
    });
  });
});
