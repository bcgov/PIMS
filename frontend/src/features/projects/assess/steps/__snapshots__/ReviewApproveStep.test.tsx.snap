// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Approve Step renders correctly 1`] = `
.c2 {
  font-size: 12px;
  margin-top: -100px;
  max-width: 16.7%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
}

.c0 {
  padding-left: 10px;
}

.c0 small {
  padding-top: 20px;
  vertical-align: middle;
}

<div
  class="ReviewApproveStep container-fluid"
>
  <form
    class=""
  >
    <h1>
      Project Application Review
    </h1>
    <div
      class="ProjectDraftForm container-fluid"
    >
      <div
        class="form-row"
      >
        <h3
          class="col-md-8"
        >
          Project Property Information
        </h3>
        <span
          class="col-md-4"
        />
      </div>
      <div
        class="col-md-10 form-row"
      >
        <label
          class="col-md-2 form-label"
        >
          Project No.
        </label>
        <div
          class="col-md-2 form-group"
        >
          <input
            class="form-control"
            disabled=""
            id="input-projectNumber"
            name="projectNumber"
            placeholder="SPP-XXXXXX"
            value="test-01"
          />
        </div>
      </div>
      <div
        class="form-row"
      >
        <div
          class="required col-md-10 form-group"
        >
          <label
            class="form-label"
            for="input-name"
          >
            Name
          </label>
          <input
            class="col-md-5 form-control"
            data-testid="project-name"
            disabled=""
            id="input-name"
            name="name"
            value="my project"
          />
        </div>
      </div>
      <div
        class="form-row"
      >
        <div
          class="col-md-10 form-group"
        >
          <label
            class="form-label"
            for="input-description"
          >
            Description
          </label>
          <textarea
            class="col-md-auto form-control"
            data-testid="project-description"
            disabled=""
            id="input-description"
            name="description"
          >
            my project description
          </textarea>
        </div>
      </div>
    </div>
    <div
      class="UpdateInfoForm container-fluid"
    >
      
      <div
        class="form-row"
      >
        <label
          class="form-label col-form-label col-md-2"
        >
          Assign Tier
        </label>
        <div
          class="required col-md-1 form-group"
        >
          <select
            class="form-select form-control"
            disabled=""
            id="input-tierLevelId"
            name="tierLevelId"
            required=""
          >
            <option
              value=""
            >
              Must Select One
            </option>
          </select>
        </div>
        <div
          class="c0"
        >
          <small>
            Tier 1: Properties with a net value less than $1M.
          </small>
        </div>
      </div>
      <div
        class="form-row"
        style="align-items: unset;"
      >
        <h3
          class="col-md-8"
        >
          Financial Information
        </h3>
      </div>
      <div
        class="form-row"
      >
        <h6
          class="col-md-12"
          style="margin: 0px 0px 1rem 0px;"
        >
          These values are for the 
          <strong>
            project
          </strong>
          , not the individual properties.
        </h6>
      </div>
      <div
        class="form-row"
      >
        <div
          class="col-md-4"
        >
          <div
            class="form-row"
          >
            <label
              class="form-label col-form-label col-md-4"
            >
              Assessed Value
            </label>
            <div
              class="required form-group col-md-6"
            >
              <div
                class="input-tooltip-wrapper"
              >
                <input
                  class="form-control input-number"
                  disabled=""
                  name="assessed"
                  placeholder=""
                  required=""
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="form-row"
          >
            <label
              class="form-label col-form-label col-md-4"
            >
              Net Book Value
            </label>
            <div
              class="required form-group col-md-6"
            >
              <div
                class="input-tooltip-wrapper"
              >
                <input
                  class="form-control input-number"
                  disabled=""
                  name="netBook"
                  placeholder=""
                  required=""
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-md-4"
        >
          <div
            class="form-row"
          >
            <label
              class="form-label col-form-label col-md-4"
            >
              Estimated Market Value
            </label>
            <div
              class="required form-group col-md-6"
            >
              <div
                class="input-tooltip-wrapper"
              >
                <input
                  class="form-control input-number"
                  disabled=""
                  name="market"
                  placeholder=""
                  required=""
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="form-row"
          >
            <label
              class="form-label col-form-label col-md-4"
            >
              Appraised Value
            </label>
            <div
              class="form-group col-md-6"
            >
              <div
                class="input-tooltip-wrapper"
              >
                <input
                  class="form-control input-number"
                  disabled=""
                  name="appraised"
                  placeholder=""
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-md-4"
        >
          <div
            class="form-row"
          >
            <label
              class="form-label col-form-label col-md-4"
            >
              Estimated Sales Costs
            </label>
            <div
              class="form-group col-md-6"
            >
              <div
                class="input-tooltip-wrapper"
              >
                <input
                  class="form-control input-number"
                  disabled=""
                  name="salesCost"
                  placeholder=""
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="form-row"
          >
            <label
              class="form-label col-form-label col-md-4"
            >
              Estimated Program Recovery Fees
            </label>
            <div
              class="form-group col-md-6"
            >
              <div
                class="input-tooltip-wrapper"
              >
                <input
                  class="form-control input-number"
                  disabled=""
                  name="programCost"
                  placeholder=""
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="form-row"
      >
        <h6
          class="col-md-12"
          style="margin: 1rem 0px;"
        >
          Please update  the property classification.
        </h6>
        <h2
          class="col-md-5"
        >
          Properties in the Project
        </h2>
      </div>
      <div
        class="container-fluid"
      >
        <div
          class="ScrollContainer"
        >
          <div
            class="table"
            role="table"
          >
            <div
              class="thead thead-light"
            >
              <div
                class="tr"
                role="row"
                style="display: flex; flex: 1 0 auto; min-width: 645px;"
              >
                <div
                  class="th reset-filter svg-btn"
                />
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                  <div
                    class="sortable-column"
                  >
                    Agency
                    <span
                      style="flex: 1 1 auto;"
                    />
                  </div>
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                  <div
                    class="sortable-column"
                  >
                    Name
                    <span
                      style="flex: 1 1 auto;"
                    />
                  </div>
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                  <div
                    class="sortable-column"
                  >
                    Civic Address
                    <span
                      style="flex: 1 1 auto;"
                    />
                  </div>
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 140 0 auto; min-width: 30px; width: 140px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                  <div
                    class="sortable-column"
                  >
                    Classification
                    <span
                      style="flex: 1 1 auto;"
                    />
                  </div>
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                  <div
                    class="sortable-column"
                  >
                    Current Zoning Code
                    <span
                      style="flex: 1 1 auto;"
                    />
                  </div>
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                  <div
                    class="sortable-column"
                  >
                    Other Projects
                    <span
                      style="flex: 1 1 auto;"
                    />
                  </div>
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 145 0 auto; min-width: 145px; width: 145px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                  <div
                    class="sortable-column"
                  >
                    Net Book Value
                    <span
                      style="flex: 1 1 auto;"
                    />
                  </div>
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 145 0 auto; min-width: 145px; width: 145px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                  <div
                    class="sortable-column"
                  >
                    Market Value
                    <span
                      style="flex: 1 1 auto;"
                    />
                  </div>
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 145 0 auto; min-width: 145px; width: 145px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                  <div
                    class="sortable-column"
                  >
                    Assessed Value
                    <span
                      style="flex: 1 1 auto;"
                    />
                  </div>
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 60 0 auto; min-width: 30px; width: 60px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                  <div
                    class="sortable-column"
                  >
                    Type
                    <span
                      style="flex: 1 1 auto;"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="tbody"
              role="rowgroup"
            >
              <div
                class="tr-wrapper"
              >
                <div
                  class="tr"
                  role="row"
                  style="display: flex; flex: 1 0 auto; min-width: 645px;"
                >
                  <div
                    class="td clickable"
                    role="cell"
                    style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: flex-start; text-align: left; align-items: flex-start; display: flex;"
                    title="Click to view property details"
                  >
                    TST
                  </div>
                  <div
                    class="td clickable"
                    role="cell"
                    style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: flex-start; text-align: left; align-items: flex-start; display: flex;"
                    title="Click to view property details"
                  >
                    123-123-123
                  </div>
                  <div
                    class="td clickable"
                    role="cell"
                    style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: flex-start; text-align: left; align-items: flex-start; display: flex;"
                    title="Click to view property details"
                  >
                    1234 Test St
                  </div>
                  <div
                    class="td clickable"
                    role="cell"
                    style="box-sizing: border-box; flex: 140 0 auto; min-width: 30px; width: 140px; justify-content: flex-start; text-align: left; align-items: flex-start; display: flex;"
                    title="Click to view property details"
                  >
                    Surplus Active
                  </div>
                  <div
                    class="td clickable"
                    role="cell"
                    style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: flex-start; text-align: left; align-items: flex-start; display: flex;"
                    title="Click to view property details"
                  />
                  <div
                    class="td"
                    role="cell"
                    style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: flex-start; text-align: left; align-items: flex-start; display: flex;"
                    title=""
                  >
                    <div
                      class="c1"
                    />
                  </div>
                  <div
                    class="td"
                    role="cell"
                    style="box-sizing: border-box; flex: 145 0 auto; min-width: 145px; width: 145px; justify-content: flex-start; text-align: left; align-items: flex-start; display: flex;"
                    title=""
                  >
                    $223
                  </div>
                  <div
                    class="td"
                    role="cell"
                    style="box-sizing: border-box; flex: 145 0 auto; min-width: 145px; width: 145px; justify-content: flex-start; text-align: left; align-items: flex-start; display: flex;"
                    title=""
                  >
                    $123
                  </div>
                  <div
                    class="td"
                    role="cell"
                    style="box-sizing: border-box; flex: 145 0 auto; min-width: 145px; width: 145px; justify-content: flex-start; text-align: left; align-items: flex-start; display: flex;"
                    title=""
                  >
                    $123
                  </div>
                  <div
                    class="td clickable"
                    role="cell"
                    style="box-sizing: border-box; flex: 60 0 auto; min-width: 30px; width: 60px; justify-content: flex-start; text-align: left; align-items: flex-start; display: flex;"
                    title="Click to view property details"
                  >
                    <svg
                      class="svg"
                    >
                      icon-lot.svg
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="tfoot tfoot-light"
            >
              <div
                class="tr"
                role="row"
                style="display: flex; flex: 1 0 auto; min-width: 645px;"
              >
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                   
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                   
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                   
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 140 0 auto; min-width: 30px; width: 140px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                   
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                   
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                   
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 145 0 auto; min-width: 145px; width: 145px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                   
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 145 0 auto; min-width: 145px; width: 145px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                   
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 145 0 auto; min-width: 145px; width: 145px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                   
                </div>
                <div
                  class="th"
                  colspan="1"
                  role="columnheader"
                  style="box-sizing: border-box; flex: 60 0 auto; min-width: 30px; width: 60px; justify-content: flex-start; text-align: left; align-items: center; display: flex;"
                >
                   
                </div>
              </div>
            </div>
          </div>
          <div
            class="table-toolbar"
          />
        </div>
      </div>
    </div>
    <h3>
      Enhanced Referral Process Exemption
    </h3>
    <p>
      <i>
        The agency has requested exemption with the below rationale:
      </i>
    </p>
    <div
      class="ProjectExemptionRequestRationale form-row"
    >
      <div
        class="col-md-10 form-group"
      >
        <label
          class="form-label"
          for="input-exemptionRationale"
        >
          Rationale
        </label>
        <textarea
          class="col-md-auto form-control"
          id="input-exemptionRationale"
          name="exemptionRationale"
          readonly=""
        >
          rationale
        </textarea>
      </div>
    </div>
    <div
      class="form-row"
    >
      <label
        class="form-label col-form-label col-md-2"
      >
        ADM Approved Exemption On
      </label>
      <div
        class="required col-md-2 form-group"
      >
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              autocomplete="off"
              class="form-control date-picker"
              id="datepicker-exemptionApprovedOn"
              name="exemptionApprovedOn"
              placeholder="--/--/----"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <h3>
      Documentation
    </h3>
    <div
      class="ProjectNotes form-row"
    >
      <div
        class="col-md-10 form-group col"
      >
        <label
          class="form-label"
          for="input-documentationNote"
        >
          Documentation Notes
        </label>
        <textarea
          class="col-md-auto form-control"
          id="input-documentationNote"
          name="documentationNote"
        />
      </div>
    </div>
    <div
      class="tasksForm reviewRequired form-row"
    >
      <div
        class="required form-group"
      >
        <div
          class="check-field"
        >
          <div
            class="form-check"
          >
            <input
              checked=""
              class="form-check-input position-static"
              disabled=""
              id="input-tasks.1.isCompleted"
              name="tasks.1.isCompleted"
              required=""
              type="checkbox"
              value="true"
            />
          </div>
          <label
            class="form-label"
            for="input-tasks.1.isCompleted"
          >
            two
          </label>
        </div>
      </div>
    </div>
    <div
      class="reviewRequired row"
    >
      <div
        class="col"
      >
        <h3>
          Appraisal
          <svg
            class="tooltip-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
            />
          </svg>
        </h3>
        <div
          class="ProjectNotes form-row"
        >
          <div
            class="col-md-12 reviewRequired form-group col"
          >
            <label
              class="form-label"
              for="input-appraisedNote"
            >
              Appraisal Notes
            </label>
            <textarea
              class="col-md-auto form-control"
              id="input-appraisedNote"
              name="appraisedNote"
            />
          </div>
          <div
            class="c2"
          >
            Please include the date the appraisal was completed
          </div>
        </div>
      </div>
    </div>
    <div
      class="reviewRequired row"
    >
      <div
        class="col"
      >
        <h3>
          First Nations Consultation
          <svg
            class="tooltip-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
            />
          </svg>
        </h3>
      </div>
    </div>
    <h3>
      Approval
    </h3>
    <div
      class="form-row"
    >
      <div
        class="required form-group"
      >
        <div
          class="check-field"
        >
          <div
            class="form-check"
          >
            <input
              checked=""
              class="form-check-input position-static"
              disabled=""
              id="input-confirmation"
              name="confirmation"
              required=""
              type="checkbox"
              value="true"
            />
          </div>
          <label
            class="form-label"
            for="input-confirmation"
          >
            My Ministry/Agency has received approval to request the property be submitted to the Enhanced Referral Program and marketed internally to other Ministry 
  and Broader Public Sector Agencies for 90 days before listing the property to be sold on the public commercial real estate market
          </label>
        </div>
      </div>
    </div>
    <div
      class="ProjectNotes form-row"
    >
      <div
        class="col-md-12 reviewRequired form-group col"
      >
        <label
          class="form-label"
          for="input-note"
        >
          Notes
        </label>
        <textarea
          class="col-md-5 form-control"
          disabled=""
          id="input-note"
          name="note"
        >
          my notes
        </textarea>
      </div>
    </div>
    <div
      class="ProjectNotes form-row"
    >
      <div
        class="col-md-12 reviewRequired form-group col"
      >
        <label
          class="form-label"
          for="input-publicNote"
        >
          Shared Notes
          <svg
            class="tooltip-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
            />
          </svg>
        </label>
        <textarea
          class="col-md-5 form-control"
          disabled=""
          id="input-publicNote"
          name="publicNote"
        >
          public note
        </textarea>
      </div>
    </div>
    <div
      class="ProjectNotes form-row"
    >
      <div
        class="col-md-12 reviewRequired form-group col"
      >
        <label
          class="form-label"
          for="input-privateNote"
        >
          Private Notes
          <svg
            class="tooltip-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
            />
          </svg>
        </label>
        <textarea
          class="col-md-5 form-control"
          disabled=""
          id="input-privateNote"
          name="privateNote"
        >
          private note
        </textarea>
      </div>
    </div>
    <div
      class="ProjectNotes form-row"
    >
      <div
        class="col-md-12 reviewRequired form-group col"
      >
        <label
          class="form-label"
          for="input-notes[22].note"
        >
          Add the following text to the ERP Notification Email
          <svg
            class="tooltip-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
            />
          </svg>
        </label>
        <textarea
          class="col-md-5 form-control"
          disabled=""
          id="input-notes[22].note"
          name="notes[22].note"
        />
      </div>
    </div>
    <div
      style="text-align: right;"
    />
  </form>
</div>
`;
